src/tr/mangatr/src/eu/kanade/tachiyomi/extension/tr/mangatr/MangaTR.kt

```diff
@@ -32,24 +36,107 @@ class MangaTR : FMReader("Manga-TR", "https://manga-tr.com", "tr") {
     // ============================== Popular ===============================
     override fun popularMangaNextPageSelector() = "div.btn-group:not(div.btn-block) button.btn-info"
 
+    override fun popularMangaSelector() = "div.row a[data-toggle]"
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        setUrlWithoutDomain(element.absUrl("href"))
+        title = element.text()
+    }
+
     // =============================== Search ===============================
-    // TODO: genre search possible but a bit of a pain
-    override fun getFilterList() = FilterList()
+    // Dynamic genre list cache
+    private var cachedGenres: List<FMReader.Genre> = emptyList()
+
+    // Filters UI: site-specific filters + dynamically fetched genres
+    override fun getFilterList(): FilterList {
+        ensureGenresLoaded()
+        val baseFilters = mutableListOf<Filter<*>>(
+            Filter.Header("Metin araması ile filtreler birlikte kullanılmaz"),
+            PublicationStatusFilter(),
+            TranslateStatusFilter(),
+            AgeRestrictionFilter(),
+            ContentTypeFilter(),
+            SpecialTypeFilter(),
+        )
+
+        if (cachedGenres.isNotEmpty()) {
+            baseFilters += FMReader.GenreList(cachedGenres)
+        } else {
+            baseFilters += Filter.Header("Türleri yüklemek için 'Sıfırla' düğmesine basın")
+            baseFilters += FMReader.GenreList(emptyList())
+        }
+
+        return FilterList(baseFilters)
+    }
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val url = "$baseUrl/arama.html".toHttpUrl().newBuilder()
-            .addQueryParameter("icerik", query)
-            .build()
-        return GET(url, headers)
+        if (query.isNotBlank()) {
+            val url = "$baseUrl/arama.html".toHttpUrl().newBuilder()
+                .addQueryParameter("icerik", query)
+                .build()
+            return GET(url, headers)
+        }
+
+        val listEndpoint = if (page <= 1) "$baseUrl/manga-list.html" else "$baseUrl/$requestPath"
+        val url = listEndpoint.toHttpUrl().newBuilder()
+        if (page > 1) {
+            url.addQueryParameter("listType", "pagination")
+            url.addQueryParameter("page", page.toString())
+        }
+
+        val genreFilter = (if (filters.isEmpty()) getFilterList() else filters)
```

Comment:
Is there any special state in the default filter list that prevents you from using an empty one?